### PR TITLE
Update to gRPC documentation for Helidon 4

### LINKED ISSUE #6461: DOCS: 4.x gRPC
## Problem Description

We need to update docs for 4.x

- [x] update client.adoc for MP
- [x] update server.adoc for MP
- [x] update client.adoc for SE
- [x] update server.adoc for SE

## Guidelines

- [x] Make sure any fixes from SE 3.x are forward ported to 4.x #7515 
- [x] Make sure any fixes from MP 3.x are forward ported to 4.x
- [x] Initial focus is on correctness
- [x] Record migration impact in this issue

## Migration Impact

Summarize incompatibilities between Helidon 3 and Helidon 4.
Summarize actions customers must take to migrate code
OK to reference external specs or documents (for example MicroProfile specs)

### PR DESCRIPTION
**Pull request by tomas-langer**
Resolves #6461 

### Documentation

Update to documentation of gRPC
- server has restricted functionallity in SE
- client is removed in SE
- both client and server are removed in MP

### CODE REVIEW

File: docs/se/tracing.adoc [hunk 1]
```diff
@@ -43,8 +43,7 @@ within a service and across multiple services. This provides insight to sequence
 which helps you identify performance and operational issues.
 Helidon includes support for distributed tracing through its own API, backed by either
 through the https://opentelemetry.io/docs/instrumentation/js/api/tracing/[OpenTelemetry API], or by
-https://opentracing.io[OpenTracing API]. Tracing is integrated with WebServer, gRPC Server,
-and Security.
+https://opentracing.io[OpenTracing API]. Tracing is integrated with WebServer, and Security.
```
Comment (ljamen) [inline docs/se/tracing.adoc:46]: integrated with WebServer and Security.

File: docs/mp/grpc/client.adoc [hunk 2]
```diff
@@ -28,261 +28,10 @@ include::{rootdir}/includes/mp.adoc[]
...
-    public CompletableFuture<String> upper(String s);
-}
-----
-
-== Examples
-link:{helidon-github-tree-url}/examples/grpc/microprofile/basic-client[Basic gRPC Client] example demonstrates  a
-simple gRPC client that invokes services from deployed gRPC server applications provided in the
-link:{helidon-github-tree-url}/examples/grpc/microprofile/basic-server-implicit[Basic gRPC Server] and
-link:{helidon-github-tree-url}/examples/grpc/microprofile/metrics[gRPC Server metrics] examples.
+gRPC is temporarily removed in Helidon, please follow issue
```
Comment (ljamen) [inline docs/mp/grpc/client.adoc:34]: Do you want me to put this in Note format for more visibility? It's at the top of the page so I think its ok as-is.

File: docs/mp/guides/tracing.adoc [hunk 3]
```diff
@@ -38,7 +38,7 @@ include::{rootdir}/includes/prerequisites.adoc[tag=prerequisites]
 Distributed tracing is a critical feature of micro-service based applications, since it traces workflow both
 within a service and across multiple services.  This provides insight to sequence and timing data for specific blocks of work,
 which helps you identify performance and operational issues.  Helidon MP includes support for distributed tracing
-through the https://opentracing.io[OpenTracing API].  Tracing is integrated with WebServer, gRPC Server,
+through the https://opentracing.io[OpenTracing API].  Tracing is integrated with WebServer,
```
Comment (ljamen) [inline docs/mp/guides/tracing.adoc:41]: remove , after WebServer. 

Tracing is integrated with WebServer and Security using....

File: docs/mp/tracing.adoc [hunk 4]
```diff
@@ -45,7 +45,7 @@ The MP OpenTracing specification is no longer required by MicroProfile. This fea
 Distributed tracing is a critical feature of micro-service based applications, since it traces workflow both
 within a service and across multiple services. This provides insight to sequence and timing data for specific blocks of work,
 which helps you identify performance and operational issues. Helidon MP includes support for distributed tracing
-through the https://opentracing.io[OpenTracing API]. Tracing is integrated with WebServer, gRPC Server,
+through the https://opentracing.io[OpenTracing API]. Tracing is integrated with WebServer,
```
Comment (ljamen) [inline docs/mp/tracing.adoc:48]: extra , again

File: docs/se/guides/tracing.adoc [hunk 5]
```diff
@@ -39,7 +39,7 @@ include::{rootdir}/includes/prerequisites.adoc[tag=prerequisites]
 Distributed tracing is a critical feature of micro-service based applications, since it traces workflow both
 within a service and across multiple services.  This provides insight to sequence and timing data for specific blocks of work,
 which helps you identify performance and operational issues.  Helidon SE includes support for distributed tracing
-through the https://opentracing.io[OpenTracing API].  Tracing is integrated with WebServer, gRPC Server,
+through the https://opentracing.io[OpenTracing API].  Tracing is integrated with WebServer,
```
Comment (ljamen) [inline docs/se/guides/tracing.adoc:42]: WebServer and Security using...

**Review by ljamen — COMMENTED** [review]
very small changes

File: docs/se/tracing.adoc (hunk 1 above)
Comment (tomas-langer) [reply docs/se/tracing.adoc:46]: You don't like the oxford comma?

**Review by ljamen — DISMISSED** [review]
Review complete, changes incorporated.

**Review by ljamen — APPROVED** [review]
Rebased.

**Pull request merged**